woocommerce/woocommerce-gutenberg-products-block
# Move `Label` component to components package

## What

This PR is based on https://github.com/woocommerce/woocommerce-blocks/pull/11214 (update/checkbox-component)
Move `Label` component to components package and update the places it is imported:

- `product-sale-badge`
- `OrderSummaryItem`
-  `ShippingRatesControlPackage`
- `FilterElementLabel`
- `FilterResetButton`
- `FilterSubmitButton`
- `LoadMoreButton`
- `Pagination`
- `SortSelect`
- `active-filters`
- `PaymentMethods`

## Why



As more developers integrate their extensions with WooCommerce Blocks, the case for reusing components grows. This starts the ball rolling by creating a new package. Rather than recreating components developers can access `wc.blocksComponents` and get the ones we need.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

### Prerequisites

1. Ensure one of the products on your site is on sale.
2. Set up at least 2 shipping rates on your site.
3. Install [Multiple Packages for WooCommerce](https://wordpress.org/plugins/multiple-packages-for-woocommerce/) and go to WooCommerce > Settings > Multiple Packages. Set the "Group by" option to Product (individual).
4. Add multiple products to your cart. Ensure you add one product more than once (so you have 2+ of a single item in the cart)
5. Add 2 reviews to a product (Go to the product page on the front end and add them via the reviews tab)
6. Install and configure [WooCommerce Stripe Payment Gateway](https://wordpress.org/plugins/woocommerce-gateway-stripe/) (Or another payment gateway that supports saved cards)

### Testing steps
1. Add the `Products (Beta)` block to a page on your site. In the page editor, ensure the `Sale` badge is showing up correctly on the on sale products.
2. Save the page and view it on the front end, ensure the sale badge is also showing there.
3. Open the Checkout block on your site. Ensure the item summary contains the list of items with the number of items in a small bubble next to them

<img width="238" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5656702/55054ed9-e044-4d93-95bf-5bbdb9e975ea">

4. Fill in your details and ensure the shipping rates are showing correctly.
5. Ensure the shipping rates show the product count correctly for packages with multiple items (including the _Product Name x 3_)

<img width="460" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5656702/165a8459-b4cf-485e-acff-50eba8cf0b38">

6. Go to the Cart Block, expand one of the shipping options in the sidebar and repeat step 5.
7. Add the `Active Filters`, `Filter by Attribute`, and `Products (Beta)` blocks to a page. On the `Filter by Attribute` block, enable the `Display product count` option. Enable the `Show 'Apply filters' button` too.
8. Ensure the product count appears in brackets on the block in the page editor on the `Filter by Attribute` block.
9. Ensure the Reset and Apply buttons show.
10. View this page in the front end and ensure the product count appears on the `Filter by Attribute` block.
11. Apply some filters using the `Filter by Attribute` block, and ensure the `Active Filters` block works and has a "Clear all" button.
12. Press the "Clear all" button and ensure it removes the active filters.
13. Add the `Reviews by Product` block to a page. Set the `STARTING NUMBER OF REVIEWS` option to 1.
14. Ensure a `Load more` button appears.
15. Ensure the "Order by" select box appears on the block and that it works correctly.
16. View this on the front end and ensure the `Load more` button appears. Click it and ensure the second review loads.
17. Add the `All Products` block to a page. Using a screen reader, focus the pagination element at the bottom of the block. Ensure it reads "Navigate to another page"
18. Ensure you're logged in to the site and go to the Checkout block.
19. Check out using Stripe, ensure the option to save the payment method is selected.
20. Add items to your cart and go back to the Checkout page. Ensure you see the saved cart and the message "Use another payment method" between the saved methods and other methods. (Skip this step if you can't get Stripe or another payment method working)

<img width="497" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5656702/68b164b5-a770-4282-9e05-0d911ef21b5e">



## Screenshots or screencast


N/A

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Make the `Label` component available in the `@woocommerce/blocks-components` package.
